packages/lib/src/vis/shared/Html.tsx

```diff
@@ -1,48 +1,42 @@
+import { assertNonNull } from '@h5web/shared';
 import { useThree } from '@react-three/fiber';
-import type { ReactNode } from 'react';
+import type { PropsWithChildren } from 'react';
 import { useLayoutEffect, useState } from 'react';
-import ReactDOM from 'react-dom';
+import ReactDOM, { createPortal } from 'react-dom';
 
 interface Props {
-  overflowCanvas?: boolean; // allow children to overflow above axes
-  container?: HTMLElement;
-  children?: ReactNode;
+  overflowCanvas?: boolean;
 }
 
-function Html(props: Props) {
-  const {
-    overflowCanvas = false,
-    container: customContainer,
-    children,
-  } = props;
+function Html(props: PropsWithChildren<Props>) {
+  const { overflowCanvas = false, children } = props;
 
   const r3fRoot = useThree((state) => state.gl.domElement.parentElement);
-  const canvasWrapper = r3fRoot?.parentElement;
+  assertNonNull(r3fRoot);
 
-  // Choose DOM container to which to append `renderTarget`
-  // (with `canvasWrapper`, `Html` children are allowed to overflow above the axes)
-  const container =
-    customContainer || (overflowCanvas ? canvasWrapper : r3fRoot);
+  const canvasWrapper = r3fRoot.parentElement;
+  assertNonNull(canvasWrapper);
 
-  const [renderTarget] = useState(() => document.createElement('div'));
+  const portalContainer = overflowCanvas ? canvasWrapper : r3fRoot;
 
-  useLayoutEffect(() => {
-    ReactDOM.render(<>{children}</>, renderTarget); // eslint-disable-line react/jsx-no-useless-fragment
-  }, [children, renderTarget]);
+  const [renderContainer] = useState(() => {
+    const div = document.createElement('div');
+    div.setAttribute('hidden', '');
```

Comment:
While I'm at it, I hide the `div` that acts as the `ReactDOM.render` container, since it is now empty.